core/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/FilterAggregationBuilder.java

```diff
@@ -89,18 +83,11 @@ protected XContentBuilder internalXContent(XContentBuilder builder, Params param
         return builder;
     }
 
-    public static FilterAggregationBuilder parse(String aggregationName, QueryParseContext context)
-            throws IOException {
-        QueryBuilder filter = context.parseInnerQueryBuilder();
-
-        if (filter == null) {
-            throw new ParsingException(null, "filter cannot be null in filter aggregation [{}]", aggregationName);
-        }
-
+    public static FilterAggregationBuilder parse(String aggregationName, QueryParseContext context) throws IOException {
+        QueryBuilder filter = context.parseInnerQueryBuilder().orElse(new MatchAllQueryBuilder());
```

Comment:
Throwing the exception was introduced by 3197f7f7d856883f66b491270648e532952ad73a and the assumption was right when we represend empty clauses with EmptyQueryBuilder.  We have a test in FilterIT (testEmptyFilter()) that checks that empty clauses should be resolved to MatchAll, which breaks when I removed EmptyQueryBuilder. So I think resolving to MatchAll is right here until we remove support for emtpy clauses altogether.